src/client.rs

```diff
@@ -152,8 +177,22 @@ impl ClientFlow {
                 return Ok(event);
             }
 
-            if let Some(event) = self.progress_receive().await? {
-                return Ok(event);
+            if self.is_waiting_for_idle_done_set() {
```

Comment:
This is the uggliest part of the PR. For the `IDLE` command to work with the current flow, I acted the same way the `imap` crate does: every time an `IDLE` event is received (or a timeout), we exit the `IDLE` mode. Lib consumers need to take care of the loop.

Reply:
Interesting. No strong opinions yet, but some thoughts:

- Not sure whether we should add this to `ClientFlow` or burden the user of `ClientFlow` with implementing this feature. I don't know the pros and cons yet. Or can we implement this as a task?
- Regarding the SansIO refactoring, I would prefer to treat time as an input and somehow pass it into `ClientFlow`. This would also make testing more feasible.
- If we want to implement this as a feature of `ClientFlow`, we should do this in a separate PR and add documentation and `flow-test` test cases.

Reply:
Alright, then I will wait for the SansIO refactoring, and submit a dedicated PR.

Reply:
Okay! If you want then I can work on this, too.

Reply:
I see that the new implem with SansIO reads and writes in parallel, so I'm curious to give it a try and see how it behaves.